src/html/generator.ml

```diff
@@ -222,6 +243,55 @@ let rec block ~config ~resolve (l : Block.t) : flow Html.elt list =
         let extra_class = [ "language-" ^ lang_tag ] in
         mk_block ~extra_class Html.pre (source (inline ~config ~resolve) c)
     | Math s -> mk_block Html.div [ block_math s ]
+    | Audio (target, content) ->
+        let content = inline ~config ~resolve content in
+        let audio src = [ Html.audio ~src ~a:[ Html.a_controls () ] [] ] in
+        let block =
+          match target with
+          | External url -> audio url
+          | Internal (Resolved uri) ->
+              let url = Link.href ~config ~resolve uri in
+              audio url
+          | Internal Unresolved ->
+              let a = Html.a_class [ "xref-unresolved" ] :: [] in
+              [ Html.span ~a content ]
+        in
+        mk_block Html.div block
+    | Video (target, content) ->
+        let content = inline ~config ~resolve content in
+        let video src = [ Html.video ~src ~a:[ Html.a_controls () ] [] ] in
+        let block =
+          match target with
+          | External url -> video url
+          | Internal (Resolved uri) ->
+              let url = Link.href ~config ~resolve uri in
+              video url
+          | Internal Unresolved ->
+              let a = [ Html.a_class [ "xref-unresolved" ] ] in
+              [ Html.span ~a content ]
+        in
+        mk_block Html.div block
+    | Image (target, alt) ->
+        let image src =
+          let alt = alt_of_inline alt in
+          let img =
+            Html.a
+              ~a:[ Html.a_href src; Html.a_class [ "img-link" ] ]
+              [ Html.img ~src ~alt () ]
+          in
+          [ img ]
+        in
+        let block =
+          match target with
+          | External url -> image url
+          | Internal (Resolved uri) ->
+              let url = Link.href ~config ~resolve uri in
+              image url
+          | Internal Unresolved ->
+              let a = [ Html.a_class [ "xref-unresolved" ] ] in
+              [ Html.span ~a (inline ~config ~resolve alt) ]
```

Comment:
For unresolved images, what do you think of displaying placeholder images like: 
![Image_not_available (1)](https://github.com/ocaml/odoc/assets/497999/a9faa8f8-d59e-4363-a645-d249ff869acf)

Reply:
When there is no alt text, yes that's a good idea! (are we allowed to use this particular image?)

When there is alt text, I'm not sure what's best!

Reply:
Personally I think it's better to let things break. This makes it mechanically discoverable (404) if you add a stub, then the page has no problem as far as machine are concerned.

Reply:
Readers also need to distinguish missing images. Perhaps the alt text could be made mandatory ?
That's what I was most worried about when I suggested to add border, padding and a caption around image blocks.

Reply:
> Readers also need to distinguish missing images.

Browsers already have their own way of notifying users about broken image sources (typically via a small icon representing a broken document), there's no need to invent anything here.

Reply:
Ah but of course this should not be an `a` this should be an `img` tag with the class `xref-unresolved` and perhaps with the `src` attribute being simply the undefined reference (so that the broken icon shows up, it does not with an empty `src`).

Reply:
Currently it's a `span` (the `a` is for attribute) which include visual feedback that something is wrong, but nothing tells the user or the browser that an image should be there. So yes, I think an `img` with broken url and the alt text as alt text (if there is one) makes sense.
I'll check if "broken url" can be made a real concept, and how the browser react!